lib/new_relic/agent/transaction.rb

```diff
@@ -520,7 +520,7 @@ def needs_middleware_summary_metrics?(name)
       end
 
       def finish
-        return unless state.is_execution_traced?
+        return unless state.is_execution_traced? && initial_segment
```

Comment:
Nice change... so this is making sure at least one segment exists. Just curious was this an issue that came up in Rails testing or an enhancement you found while in code?

Reply:
This fix was driven by the `rails` multiverse suite. Related to https://github.com/newrelic/newrelic-ruby-agent/issues/2263, this change was needed to eliminate an error that showed up only in the logs but did not cause test assertions to fail.